packages/component/src/SendBox/TextBox.js

```diff
@@ -124,7 +126,13 @@ const TextBox = ({ className }) => {
   const sendBoxString = localize('TEXT_INPUT_ALT');
   const typeYourMessageString = localize('TEXT_INPUT_PLACEHOLDER');
 
-  const handleChange = useCallback(({ target: { value } }) => setTextBoxValue(value), [setTextBoxValue]);
+  const handleChange = useCallback(
+    ({ target: { value } }) => {
+      value = setEmoji(value);
+      setTextBoxValue(value);
+    },
+    [setEmoji, setTextBoxValue]
```

Comment:
I don't understand using the setter functions here as dependencies of the callback. Are we expecting those functions to change at some point that would need us to recalculate the handleChange thing?

Reply:
It's a 'good practice' that Web Chat uses throughout the repo

https://stackoverflow.com/questions/57180731/react-is-usestate-hooks-setter-able-change-during-a-component-life